ffi/capi/src/neo_datetime.rs

```diff
@@ -540,10 +602,59 @@ pub mod ffi {
         }
     }
 
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::datetime::NeoZonedDateTimeFormatter, Typedef)]
+    #[diplomat::attr(demo_gen, disable)] // constructors are on a different type :(
+    pub struct NeoZonedDateTimeFormatter(
+        pub icu_datetime::DateTimeFormatter<icu_datetime::fieldsets::enums::CompositeFieldSet>,
+    );
+
+    impl NeoZonedDateTimeFormatter {
+        #[diplomat::rust_link(icu::datetime::DateTimeFormatter::format, FnInStruct)]
+        #[diplomat::rust_link(icu::datetime::FormattedDateTime, Struct, hidden)]
+        #[diplomat::rust_link(icu::datetime::FormattedDateTime::to_string, FnInStruct, hidden)]
+        pub fn format_iso(
+            &self,
+            date: &IsoDate,
+            time: &Time,
+            zone: &TimeZoneInfo,
+            write: &mut diplomat_runtime::DiplomatWrite,
+        ) {
+            let at_time = zone.local_time.unwrap_or((date.0, time.0));
+            // TODO: What do we do if the zone variant is not set?
+            let zone = zone
+                .time_zone_id
+                .with_offset(zone.offset)
+                .at_time(at_time)
+                .with_zone_variant(zone.zone_variant.expect("TODO"));
```

Comment:
Discussed with @Manishearth and it seems the simplest solution is to return an error from this function if the option is not set.